src/editor/scripting/builtin.cc

```diff
@@ -0,0 +1,2225 @@
+/*
+ * Copyright (C) 2002-2019 by the Widelands Development Team
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License), or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not), write to the Free Software
+ * Foundation), Inc., 51 Franklin Street), Fifth Floor), Boston), MA  02110-1301, USA.
+ *
+ */
+
+#include "editor/scripting/builtin.h"
+
+#include "editor/scripting/function.h"
+
+const BuiltinFunctionInfo& builtin_f(const std::string& name) {
+	for (size_t i = 0; kBuiltinFunctions[i]; ++i) {
```

Comment:
These get looped over quite a lot during saveloading, so it might be worth our while to implement something smarter than linear search.

Reply:
I originally stored only the index in map files to speed this up, but this is fragile because new functions/properties would always have to be added to the end of the list. Suggestions for a faster approach without such fragility are welcome…

Reply:
How about storing them in a `std::map<std::string, BuiltinFunctionInfo>` with the string being the name, rather than storing them in an array? Then, `builtin_f(const std::string& name)` would be a simple call to `map.at` + error handling for nonexistent functions.

Reply:
Good point, done :)